### PR TITLE
ticket 7 first commit

### PR DESCRIPTION
**Pull request by TomMirfin**
ticket 7 pull request

### CODE REVIEW

File: controller/comments.controller.js [hunk 1]
```diff
@@ -1,3 +1,20 @@
+const { postComment } = require("../model/comments-model");
+
+exports.postCommentByID = (req, res, next) => {
+  const newComment = req.body;
+
+  const { article_id } = req.params;
+
+  postComment(newComment, article_id)
+    .then((comment) => {
+      console.log(comment, "<--- comment in controller");
```
Comment (mezzDavies) [inline controller/comments.controller.js:10]: Please remove console.logs

File: controller/comments.controller.js [hunk 2]
```diff
@@ -1,3 +1,20 @@
...
+      res.status(201).send(comment);
+    })
+    .catch((err) => {
+      console.log(err.code);
```
Comment (mezzDavies) [inline controller/comments.controller.js:14]: 👀

File: model/comments-model.js [hunk 3]
```diff
@@ -1,5 +1,19 @@
 const db = require("../db/connection.js");
 
+exports.postComment = ({ username, body }, article_id) => {
+  const articleNumber = parseInt(article_id);
```
Comment (mezzDavies) [inline model/comments-model.js:4]: Is this step required?

File: __tests__/endpoint.test.js [hunk 4]
```diff
@@ -120,6 +120,38 @@ describe("/api/articles", () => {
...
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body).toMatchObject({
+          body: "Hello this is my new comment",
+          author: "rogersop",
+        });
+      });
+  });
+  test("404: If a new comment is posted with a username that doesn't exist an error will be returned", () => {
```
Comment (mezzDavies) [inline __tests__/endpoint.test.js:140]: Great test 👍

**Review by mezzDavies — CHANGES_REQUESTED** [review]
This is a decent start but you need to add a few more test here:

What if the req body (eg the comment) has missing or incomplete properties thta must be included for it to be added?

Also the other 2 standard tests when using an ID - what if it is an invalid id eg not a number but the string "banana" or what if it is valid but doesn't exist eg 99999?
99999

Also I can see you still haven't quite grasped when to use a 400 vs a 404 - have a [look here](https://notes.northcoders.com/courses/js-back-end/error-handling#2-different-errors-and-responses-that-should-be-sent) for a refresh 🙂

File: __tests__/endpoint.test.js [hunk 5]
```diff
@@ -174,3 +174,65 @@ describe("/api/articles/:article_id/comments", () => {
...
+    const newComment = {
+      username: "rogersop",
+      body: "Hello this is my new comment",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body).toMatchObject({
```
Comment (Hard-Luck) [inline __tests__/endpoint.test.js:188]: should have a comment id on it

**Review by Hard-Luck — APPROVED** [review]
Tiny change then good to merge, good work 👍

**Pull request merged**